djangocms_versioning/cms_config.py

```diff
@@ -73,6 +71,22 @@ def handle_versioning_setting(self, cms_config):
             # Checks passed. Add versionable to our master list
             self.versionables.append(versionable)
 
+    def handle_versioning_add_to_confirmation_context_setting(self, cms_config):
+        """
+        Check versioning_add_to_confirmation_context has been correctly set
+        and add it to the master dict if all is ok
+        """
+        add_to_context = cms_config.versioning_add_to_confirmation_context
+        supported_keys = ['unpublish']
+        for key, value in add_to_context.items():
+            if key not in supported_keys:
+                raise ImproperlyConfigured(
+                    "{!r} is not a supported dict key in the versioning_add_to_confirmation_context setting".format(key)
+                )
+            if key not in self.add_to_context:
```

Comment:
Can be removed if `self.add_to_context = {}` gets changed to `self.add_to_context = defaultdict(OrderedDict)``